File: backend/src/UnCRM.Api/Domain/Services/Classes/PessoaService.cs
using AutoMapper;
using UnCRM.Api.Contract.Pessoa;
using UnCRM.Api.Domain.Models;
using UnCRM.Api.Domain.Repository.Interfaces;
using UnCRM.Api.Domain.Services.Interfaces;
using UnCRM.Api.Exceptions;

namespace UnCRM.Api.Domain.Services.Classes
{
    public class PessoaService(IPessoaRepository repository, IMapper mapper) : IPessoaService
    {
        public async Task<PessoaResponseContract> Adicionar(PessoaRequestContract request)
        {
            await request.Validar();

            var pessoa = mapper.Map<Pessoa>(request);
            pessoa.DataCadastro = DateTime.Now;

            pessoa = await repository.Adicionar(pessoa);

            return mapper.Map<PessoaResponseContract>(pessoa);
        }


        public async Task<PessoaResponseContract> Atualizar(long id, PessoaRequestContract request)
        {
            await request.Validar();

            var entidade = await repository.Obter(id) 
                ?? throw new NotFoundException("Pessoa não encontrada.");

            mapper.Map(request, entidade);

            await repository.Atualizar(entidade);

            return mapper.Map<PessoaResponseContract>(entidade);
        }


        public async Task Inativar(long id)
        {
            var pessoa = await repository.Obter(id) 
                ?? throw new NotFoundException("Pessoa não encontrada para inativação.");

            await repository.Deletar(pessoa);
        }

        public async Task<IEnumerable<PessoaResponseContract>> ObterTodos()
        {
            var pessoas = await repository.Obter();

            return pessoas.Select(mapper.Map<PessoaResponseContract>);
        }

        public async Task<PessoaResponseContract> ObterPorId(long id)
        {
            var pessoa = await repository.Obter(id) 
                ?? throw new NotFoundException("Pessoa não encontrada.");

            return mapper.Map<PessoaResponseContract>(pessoa);
        }
    }
}

File: frontend/uncrm/src/services/pessoaService.js
import api from "./api";

function cadastrar(pessoa) {
  return new Promise((resolve, reject) => {
    return api
      .post(`/pessoas`, pessoa)
      .then((response) => resolve(response))
      .catch((error) => reject(error));
  });
}
function obterTodos() {
  return new Promise((resolve, reject) => {
    return api
      .get("/pessoas")
      .then((response) => resolve(response))
      .catch((error) => reject(error));
  });
}
function obterPorId(id) {
  return new Promise((resolve, reject) => {
    return api
      .get(`/pessoas/${id}`)

      .then((response) => resolve(response))
      .catch((error) => reject(error));
  });
}
function atualizar(pessoa) {
  return new Promise((resolve, reject) => {
    return api
      .put(`/pessoas/${pessoa.id}`, pessoa)
      .then((response) => resolve(response))
      .catch((error) => reject(error));
  });
}
function inativar(pessoa) {
  return new Promise((resolve, reject) => {
    return api
      .delete(`/pessoas/${pessoa.id}`, pessoa)
      .then((response) => resolve(response))
      .catch((error) => reject(error));
  });
}

export default {
  obterTodos,
  obterPorId,
  cadastrar,
  atualizar,
  inativar,
};
